### PR TITLE
Added markdown preview for event email feature

### LINKED ISSUE #60: Add a preview view to the event emails
We are having markdown support now for event mails. It would be nice to allow users to see a preview of the email the are writing.

### PR DESCRIPTION
**Pull request by nikhilgupta1211**
Created new.js.erb for the ajax request and preview for the response

Fixes #60

### CODE REVIEW

**Comment by nikhilgupta1211** [comment]
image

image

**Comment by nikhilgupta1211** [comment]
@ChrisBr @bgeuken I have completed the markdown preview feature please have a look at it :)

File: app/controllers/event_emails_controller.rb [hunk 1]
```diff
@@ -1,7 +1,12 @@
 class EventEmailsController < InheritedResources::Base
+  respond_to :html, :js
```
Comment (ChrisBr) [inline app/controllers/event_emails_controller.rb]: Why is that necessary?

File: app/assets/javascripts/event_emails.js.coffee [hunk 2]
```diff
@@ -27,3 +27,17 @@ $("input[type='checkbox']").each ->
 
 $('#state-menu').click (event) ->
   event.stopPropagation()
+
+# To limit the ajax call to only when the content of body is changed
+flag = true
```
Comment (ChrisBr) [inline app/assets/javascripts/event_emails.js.coffee]: Drop it, I don't think this will be an issue and makes the code more complicated

File: app/views/event_emails/new.html.haml [hunk 3]
```diff
@@ -13,7 +13,18 @@
...
+          %a{"aria-controls" => "body", "data-toggle" => "tab", :href => "#body", :role => "tab"} Body
+        %li{:role => "presentation"}
+          %a{"aria-controls" => "preview", "data-toggle" => "tab", :href => "#preview", :role => "tab", "data-event-id" => "#{@event.id}"} Preview
+      .tab-content
+        #body.tab-pane.active{:role => "tabpanel"}
+          = f.input :body, as: :text, hint: 'This field supports markdown', input_html: {rows: 12}, label: false
+        #preview.tab-pane{:role => "tabpanel"}
+          .panel.panel-default
+            .panel-body#preview_screen
     %a{:href => "https://daringfireball.net/projects/markdown/syntax"}
```
Comment (ChrisBr) [inline app/views/event_emails/new.html.haml]: Hm, doesn't this belong to the #body tab?

File: app/views/event_emails/new.html.haml [hunk 4]
```diff
@@ -13,7 +13,18 @@
...
+        %li.active{:role => "presentation"}
+          %a{"aria-controls" => "body", "data-toggle" => "tab", :href => "#body", :role => "tab"} Body
+        %li{:role => "presentation"}
+          %a{"aria-controls" => "preview", "data-toggle" => "tab", :href => "#preview", :role => "tab", "data-event-id" => "#{@event.id}"} Preview
+      .tab-content
+        #body.tab-pane.active{:role => "tabpanel"}
+          = f.input :body, as: :text, hint: 'This field supports markdown', input_html: {rows: 12}, label: false
+        #preview.tab-pane{:role => "tabpanel"}
+          .panel.panel-default
+            .panel-body#preview_screen
```
Comment (ChrisBr) [inline app/views/event_emails/new.html.haml]: What about adding some "loading" message?

File: app/controllers/event_emails_controller.rb (hunk 1 above)
Comment (nikhilgupta1211) [inline app/controllers/event_emails_controller.rb]: Yes sorry I forgot to remove this

File: app/assets/javascripts/event_emails.js.coffee (hunk 2 above)
Comment (nikhilgupta1211) [inline app/assets/javascripts/event_emails.js.coffee]: ok :)

File: app/views/event_emails/new.html.haml (hunk 4 above)
Comment (nikhilgupta1211) [inline app/views/event_emails/new.html.haml]: Adding it.

File: app/views/event_emails/new.html.haml (hunk 3 above)
Comment (nikhilgupta1211) [inline app/views/event_emails/new.html.haml]: yes

**Comment by nikhilgupta1211** [comment]
@ChrisBr I have made the chages :)

File: app/assets/javascripts/event_emails.js.coffee [hunk 5]
```diff
@@ -27,3 +27,10 @@ $("input[type='checkbox']").each ->
 
 $('#state-menu').click (event) ->
   event.stopPropagation()
+
+$('a[href=#preview]').click ->
+  $("#preview_screen").html("<h3>Loading...</h3>")
```
Comment (nikhilgupta1211) [inline app/assets/javascripts/event_emails.js.coffee]: image
